// @cora/cora-component-library/src/components/Modal/index.tsx
import { FC, useState } from "react"
import { Modal as RBModal } from "react-bootstrap"
import { Button } from "../.."

export interface ModalProps {
  children: React.ReactNode
  title: string
  centered?: boolean
  buttonText: string
  primaryText?: string
  secondaryText?: string
  primaryOnClick?: () => void
  secondaryOnClick?: () => void
}

export const Modal: FC<ModalProps> = ({
  children,
  title,
  centered,
  buttonText,
  primaryText,
  secondaryText,
  primaryOnClick,
  secondaryOnClick,
}: ModalProps) => {
  const [show, setShow] = useState(false)

  const handleClose = () => setShow(false)
  const handleShow = () => setShow(true)

  return (
    <div className="cora-modal">
      <Button type="primary" onClick={handleShow}>
        {buttonText}
      </Button>

      <RBModal
        show={show}
        onHide={handleClose}
        aria-labelledby={title}
        centered={centered}
      >
        <RBModal.Header closeButton className="header">
          <RBModal.Title className="title">{title}</RBModal.Title>
        </RBModal.Header>
        <RBModal.Body>{children}</RBModal.Body>
        <RBModal.Footer className="footer gap-2 d-md-flex justify-content-md-end">
          <Button
            type="secondary"
            onClick={() => {
              secondaryOnClick && secondaryOnClick()
              handleClose()
            }}
          >
            {secondaryText || "Close"}
          </Button>
          <Button
            type="primary"
            onClick={() => {
                primaryOnClick && primaryOnClick()
              handleClose()
            }}
          >
            {primaryText || "Save"}
          </Button>
        </RBModal.Footer>
      </RBModal>
    </div>
  )
}

export default Modal
