// @cora/cora-component-library/src/components/Dropdown/Dropdown.stories.tsx
import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import Dropdown from "./";

export default {
  title: "Core Components/Dropdown",
  component: Dropdown,
  argTypes: {},
} as ComponentMeta<typeof Dropdown>;

const Template: ComponentStory<typeof Dropdown> = (args) => (
  <Dropdown {...args} />
);

const listItems = [
  { text: "item 1", onClick: () => alert("Item 1 Clicked"), key: "1" },
  { text: "item 2", onClick: () => alert("Item 2 Clicked"), key: "2" },
  { text: "item 3", onClick: () => alert("Item 3 Clicked"), key: "3" },
];

export const Square = Template.bind({});
Square.args = {
  label: "Dropdown",
  color: "primary",
  type: "square",
  listItems: listItems,
};

export const Round = Template.bind({});
Round.args = {
  label: "DD",
  color: "secondary",
  type: "round",
  listItems: listItems,
};

export const Ellipsis = Template.bind({});
Ellipsis.args = {
  label: "DD",
  color: "transparent",
  type: "ellipsis",
  listItems: listItems,
};
