// @cora/cora-component-library/src/components/Pill/index.tsx
import React from "react";
import { css } from "react-select/dist/declarations/src/components/SingleValue";
import styled from "styled-components";

export interface IPillProps {
  text: string;
  color: "red" | "green" | "purple";
  isActive?: boolean;
  icon?: React.ReactNode;
}

const PillContainer = styled.div<{ color: string; isActive: boolean }>`
  border: 1px solid var(--cora-${({ color }) => color}-light);
  color: var(--cora-${({ color }) => color}-light);
  border-radius: 30px;
  height: 30px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  user-select: none;
  cursor: pointer;
  padding: 0 10px;

  ${({ isActive }) =>
    !isActive &&
    `
    background-color: var(--cora-gray-light);
    border: 1px solid var(--cora-gray-light);
    color: var(--cora-gray);
    `};
`;

const PillIcon = styled.div`
  margin-right: 5px;
`;

export const Pill: React.FC<IPillProps> = ({ text, color, isActive, icon }) => {
  const [isActiveState, setIsActive] = React.useState<boolean>(false);

  return (
    <PillContainer
      onClick={() => setIsActive(!isActiveState)}
      color={color}
      isActive={isActiveState}
    >
      <PillIcon>{icon}</PillIcon>
      {text}
    </PillContainer>
  );
};

export default Pill;
