// @cora/cora-component-library/src/components/Slider/sliderStyles.tsx
import styled from "styled-components";

export const SliderContainer = styled.div`
  .e-control-wrapper {
    &.e-slider-container {
      .e-slider {
        //handle styling
        .e-handle {
          background-color: var(--cora-red);
          border: 1px solid var(--cora-red);
          width: 32px;
          height: 32px;
          border-radius: 50%;

          &.e-tab-handle {
            opacity: 0.8;
            box-shadow: 0 0 0 3px var(--cora-red-very-light);
          }
        }
        //bar styling
        .e-range {
          background-color: var(--cora-red-light);
        }
      }

      &.e-horizontal {
        //bar position
        .e-handle {
          margin-left: -16px;
          top: calc(50% - 16px);
        }
        //bar + coloured bar height
        .e-slider-track,
        .e-range {
          height: 8px;
        }
      }

      .e-scale {
        //tick font size
        .e-tick {
          .e-tick-value {
            font-size: 1rem;
          }
        }
        //position of the tick under the slider bar
        &.e-h-scale {
          .e-tick {
            .e-tick-value {
              &.e-tick-after {
                bottom: -25px;
              }
            }
          }
        }
      }
    }
  }
`;

export const SliderHeaderContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const SliderInputContainer = styled.div`
  width: 160px;
  display: flex;
  justify-content: end;
  input {
    width: 80px;
    margin-left: 5px;
  }
`;

export const SliderHelpTextContainer = styled.div`
  margin-top: 10px;
`;
