// @cora/cora-component-library/src/components/Slider/slider.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";

import Slider from ".";

export default {
  title: "Forms/Slider",
  component: Slider,
  argTypes: {},
} as ComponentMeta<typeof Slider>;

const Template: ComponentStory<typeof Slider> = (args) => {
  return <Slider {...args} />;
};

export const Default = Template.bind({});
Default.args = {
  label: "Default Slider",
  type: "Default",
  minValue: 1,
  maxValue: 10,
  value: 5,
  step: 1,
  showTicks: false,
  helpText: "Help text for slider component.",
  onChange: (e) => console.log(e),
};

export const MinRange = Template.bind({});
MinRange.args = {
  label: "MinRange Slider Example",
  type: "MinRange",
  minValue: 1,
  maxValue: 10,
  value: 5,
  step: 1,
  showTicks: false,
  helpText: "Help text for slider component.",
  onChange: (e) => console.log(e),
};

export const Range = Template.bind({});
Range.args = {
  label: "Range Slider Example",
  type: "Range",
  minValue: 1,
  maxValue: 10,
  value: [2, 7],
  step: 1,
  showTicks: true,
  helpText: "Help text for slider component.",
  onChange: (e) => console.log(e),
};
