// @cora/cora-component-library/src/components/Repeater/repeater.stories.tsx
import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { Column, Row } from "./index";

import Repeater from ".";

export default {
  title: "Core Components/Repeater",
  component: Repeater,
  argTypes: {},
} as ComponentMeta<typeof Repeater>;

const Template: ComponentStory<typeof Repeater> = (args) => (
  <Repeater {...args} />
);

const columnTemplate: Column[] = [
  {
    title: "Data point to be collected",
    type: "Text",
  },
  {
    title: "Input Type",
    type: "List",
    listItems: [
      { value: "1", label: "Ireland" },
      { value: "2", label: "France" },
      { value: "3", label: "Spain" },
    ],
  },
];

const rowData: Row[] = [
  { checked: false, data: ["Entity name", "1"] },
  { checked: true, data: ["Entity Start Date", "3"] },
  { checked: false, data: ["Entity End Date", "2"] },
  { checked: false, data: ["Entity cost", "1"] },
  { checked: false, data: ["Entity owner", "2"] },
  { checked: false, data: ["Entity owner email", "3"] },
];

export const Default = Template.bind({});
Default.args = {
  columnTemplate: columnTemplate,
  rows: rowData,
  showCheckbox: true,
  showReorder: true,
  showPosition: true,
  showExpand: true,
  expandOnClick: () => alert("Expand Clicked"),
  onCreateOption: (e, rowIndex) => {
    console.log(e);
  },
};

const columnTemplateBasic: Column[] = [
  {
    title: "Region",
    type: "List",
    listItems: [
      { value: "1", label: "Ireland" },
      { value: "2", label: "France" },
      { value: "3", label: "Spain" },
    ],
  },
];

const rowDataBasic: Row[] = [];

//{ checked: false, data: ["1"] },
//{ checked: false, data: ["1"] },

export const Basic = Template.bind({});
Basic.args = {
  columnTemplate: columnTemplateBasic,
  rows: rowDataBasic,
  showCheckbox: false,
  showPosition: false,
  showReorder: false,
  showExpand: true,
  showAdd: true,
  expandOnClick: () => alert("Expand Clicked"),
};

const columnTemplateRepeaterInput: Column[] = [
  {
    title: "Region",
    type: "Input",
  },
];

const rowDataRepeaterInput: Row[] = [];

export const RepeaterInput = Template.bind({});
RepeaterInput.args = {
  columnTemplate: columnTemplateRepeaterInput,
  rows: rowDataRepeaterInput,
  showCheckbox: false,
  showPosition: false,
  showReorder: false,
  showExpand: true,
  showAdd: true,
  expandOnClick: () => alert("Expand Clicked"),
  onCreateOption: (e, rowIndex) => {
    console.log(e);
  },
};
