// @cora/cora-component-library/src/components/SelectAutoComplete/index.tsx
import React from "react";
import { Form } from "react-bootstrap";
import Select from "react-select";
import { IFormField } from "../Common/Interfaces/IFormField";
import Creatable from "react-select/creatable";

export interface ISelectAutoCompleteProps extends IFormField {
  options: any;
  allowMultiselect?: boolean;
  allowCreate?: boolean;
  onCreateOption: (e: any) => void;
}

export const SelectAutoComplete: React.FC<ISelectAutoCompleteProps> = ({
  id,
  options,
  label,
  validation,
  helpText,
  allowMultiselect,
  allowCreate,
  onCreateOption,
}) => {
  return (
    <Form.Group id={id} className="mb-3">
      <Form.Label>{label}</Form.Label>
      {allowCreate && (
        <Creatable
          onCreateOption={(e) => onCreateOption(e)}
          options={options}
          isMulti={allowMultiselect}
        />
      )}
      {!allowCreate && <Select options={options} isMulti={allowMultiselect} />}
      {validation && (
        <Form.Control.Feedback
          type={validation.isInvalid ? "invalid" : "valid"}
        >
          {validation.validationMessage}
        </Form.Control.Feedback>
      )}
      {helpText && <Form.Text className="text-muted">{helpText}</Form.Text>}
    </Form.Group>
  );
};

export default SelectAutoComplete;
