// @cora/cora-component-library/src/components/SelectAutoComplete/selectAutoComplete.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";

import SelectAutoComplete from ".";

export default {
  title: "Forms/SelectAutoComplete",
  component: SelectAutoComplete,
  argTypes: {},
} as ComponentMeta<typeof SelectAutoComplete>;

const Template: ComponentStory<typeof SelectAutoComplete> = (args) => {
  return <SelectAutoComplete {...args} />;
};

const options = [
  { value: "1", label: "Ireland" },
  { value: "2", label: "France" },
  { value: "3", label: "Spain" },
];

const validation = {
  isValid: false,
  isInvalid: true,
  validationMessage: "Validation message",
};

export const Single = Template.bind({});
Single.args = {
  options: options,
  label: "Autocomplete Select",
  validation: validation,
  helpText: "Help text",
};

export const Multi = Template.bind({});
Multi.args = {
  options: options,
  label: "Autocomplete Select",
  validation: validation,
  allowMultiselect: true,
  helpText: "Help text",
};

export const Create = Template.bind({});
Create.args = {
  options: options,
  label: "Autocomplete Select",
  validation: validation,
  allowMultiselect: true,
  helpText: "Help text",
  allowCreate: true,
  onCreateOption: (e) => {
    alert(e);
  },
};
