// @cora/cora-component-library/src/components/Buttons/ButtonMultiline/index.tsx
import React, { FC, ReactNode } from "react";
import { Button as RBButton } from "react-bootstrap";

export interface ButtonProps {
  children?: ReactNode;
  type?: "primary" | "secondary";
  align?: "start" | "end";
  subText?: string | "";
  onClick?: () => void;
  disabled?: boolean;
}

export const ButtonMultiline: FC<ButtonProps> = ({
  children,
  type,
  align,
  subText,
  onClick,
  disabled,
}: ButtonProps) => {
  let varient = "primary";
  let classes = [];

  switch (type) {
    case "primary":
      classes.push("btn btn-primary");
      break;
    case "secondary":
      classes.push("cora-button-secondary");
      varient = `outline-${varient}`;
      break;
  }

  if (align === "end") {
    classes.push("align-items-end");
  }

  return (
    <RBButton
      variant={varient}
      className={`cora-button-multiline ${classes.join(" ")}`}
      onClick={onClick}
      disabled={disabled}
    >
      <span>{children}</span>
      <span className="subText">{subText}</span>
    </RBButton>
  );
};

export default ButtonMultiline;
