// @cora/cora-component-library/src/components/Spinner/spinner.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";

import Spinner from ".";

export default {
  title: "Core Components/Spinner",
  component: Spinner,
  argTypes: {},
} as ComponentMeta<typeof Spinner>;

const Template: ComponentStory<typeof Spinner> = (args) => {
  return (
    <div style={{ width: "100%", height: "600px" }}>
      <Spinner {...args} />
    </div>
  );
};

export const Default = Template.bind({});
Default.args = {
  showSpinner: true,
  text: "Loading",
};

const TemplateBackground: ComponentStory<typeof Spinner> = (args) => {
  return (
    <div style={{ width: "100%", height: "600px" }}>
      <img
        width={500}
        style={{
          position: "fixed",
          top: "50%",
          left: "50%",
          transform: "translate(-50%, -50%)",
        }}
        src="https://www.rd.com/wp-content/uploads/2021/04/GettyImages-988013222-scaled-e1618857975729.jpg"
      ></img>
      <Spinner {...args} />
    </div>
  );
};

export const Background = TemplateBackground.bind({});
Background.args = {
  showSpinner: false,
  text: "Loading",
};
