// @cora/cora-component-library/src/components/CheckRadioGroup/checkRadioGroup.stories.tsx
import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { Validation } from "../Validation/validation";

import CheckboxGroup from ".";
import { ICheckRadioProps } from "../CheckRadio";

export default {
  title: "Forms/CheckRadioGroup",
  component: CheckboxGroup,
  argTypes: {},
} as ComponentMeta<typeof CheckboxGroup>;

const Template: ComponentStory<typeof CheckboxGroup> = (args) => {
  return <CheckboxGroup {...args} />;
};

const validation1: Validation = {
  isValid: false,
  isInvalid: false,
  validationMessage: "Validation Message",
};

const validation2: Validation = {
  isValid: false,
  isInvalid: false,
  validationMessage: "Validation Message",
};

const validation3: Validation = {
  isValid: false,
  isInvalid: false,
  validationMessage: "Validation Message",
};

const checkboxGroupItems: ICheckRadioProps[] = [
  {
    label: "Item 1",
    validation: validation1,
  },
  {
    label: "Item 2",
    helpText: "Help tect for item 2",
    validation: validation2,
  },
  {
    label: "Item 3",
    validation: validation3,
  },
];

export const Checkbox = Template.bind({});
Checkbox.args = {
  checkboxGroupItems: checkboxGroupItems,
  name: "test",
};

export const Radio = Template.bind({});
Radio.args = {
  checkboxGroupItems: checkboxGroupItems,
  name: "test",
  type: "radio",
};
