// @cora/cora-component-library/src/components/NavBar/index.tsx
import React from 'react';
import { Nav } from 'react-bootstrap';
import clsx from 'clsx';
import NavBarOverflowWrapper from './navBarOverflowWrapper';

export interface NavItemProps {
  href: string;
  title: string;
}

export interface NavBarProps {
  navItems: NavItemProps[];
  component?: any;
  aditionalComponentProps?: Record<string, unknown>;
  activeRoute?: string;
}

const formatToCamelCase = (input: string) =>
  input.toLowerCase().replace(/\s+(\w)?/gi, (match, letter) => {
    return letter.toUpperCase();
  });

export const NavBar: React.FC<NavBarProps> = ({ navItems, component: Component, activeRoute, aditionalComponentProps }) => {
  return (
    <NavBarOverflowWrapper component={Component} aditionalComponentProps={aditionalComponentProps}>
      {navItems.map(({ title, href }) => {
        const formattedTitle = formatToCamelCase(title);
        const active = activeRoute === href;
        return Component ? (
          <Component href={href} key={formattedTitle} {...aditionalComponentProps}>
            <Nav.Link data-targetid={formattedTitle} className={clsx('nav-item me-3', active && 'active')}>
              {title}
            </Nav.Link>
          </Component>
        ) : (
          <Nav.Link key={formattedTitle} data-targetid={formattedTitle} className={clsx('nav-item me-3', active && 'active')} href={href}>
            {title}
          </Nav.Link>
        );
      })}
    </NavBarOverflowWrapper>
  );
};

export default NavBar;
