// @cora/cora-component-library/src/components/NotificationCard/notificationCard.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";

import NotificationCard from ".";

export default {
  title: "Page Components/NotificationCard",
  component: NotificationCard,
  argTypes: {},
} as ComponentMeta<typeof NotificationCard>;

const Template: ComponentStory<typeof NotificationCard> = (args) => {
  return <NotificationCard {...args} />;
};

export const Default = Template.bind({});
Default.args = {
  cardType: "calander",
  headerText: "Data Updated",
  bodyText: "Entity data, Increase EV Fleet has been updated",
  detailsButtonText: "DETAILS",
  detailsButtonOnClick: () => alert("Details Button Clicked"),
  snoozeButtonText: "SNOOZE",
  deteteButtonText: "DELETE",
  deleteButtonOnClick: () => alert("Delete Button Clicked"),
};

const TemplateContainer: ComponentStory<typeof NotificationCard> = (args) => {
  return (
    <div style={{ width: "450px" }}>
      <NotificationCard {...args} />
    </div>
  );
};

export const Container = TemplateContainer.bind({});
Container.args = {
  cardType: "calander",
  headerText: "Data Updated",
  bodyText:
    "Entity data, Increase EV Fleet has been updated. Entity data, Increase EV Fleet has been updated",
  detailsButtonText: "DETAILS",
  detailsButtonOnClick: () => alert("Details Button Clicked"),
  snoozeButtonText: "SNOOZE",
  deteteButtonText: "DELETE",
  deleteButtonOnClick: () => alert("Delete Button Clicked"),
};
