// @cora/cora-component-library/src/components/ProgressGroup/progressGroup.stories.tsx
import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import ProgressGroupItem, { ProgressGroupItemProps } from "./progressGroupItem";

import ProgressGroup from ".";

export default {
  title: "Page Components/ProgressGroup",
  component: ProgressGroup,
  argTypes: {},
} as ComponentMeta<typeof ProgressGroup>;

const Template: ComponentStory<typeof ProgressGroup> = (args) => (
  <ProgressGroup {...args} />
);

const data: ProgressGroupItemProps[] = [
  {
    percentage: 65,
    onEdit: () => {
      alert("Edit Clicked - Enterprise");
    },
    label: "Enterprise",
    info: {
      title: "Enterprise Details",
      description:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean euismod bibendum laoreet. ",
      actionText: "ACTION",
      onAction: () => {
        alert("Enterprise Action");
      },
    },
    isActive: true,
    isFirst: true,
  },
  {
    percentage: 45,
    onEdit: () => {
      alert("Edit Clicked - Business Details");
    },
    label: "Business Details",
    isComplete: true,
    info: {
      title: "Business Details",
      description:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean euismod bibendum laoreet. ",
      onAction: () => {
        alert("Business Action");
      },
    },
  },
  {
    percentage: 0,
    onEdit: () => {
      alert("Edit Clicked - Currency Rules");
    },
    label: "Currency Rules",
    info: {
      title: "Currency Details",
      description:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean euismod bibendum laoreet. ",
      actionText: "ACTION",
      onAction: () => {
        alert("Currency Action");
      },
    },
  },
];

const groupItems = [
  <ProgressGroupItem {...data[0]} />,
  <ProgressGroupItem {...data[1]} />,
  <ProgressGroupItem {...data[2]} />,
];

export const Default = Template.bind({});
Default.args = {
  children: groupItems,
};
