// @cora/cora-component-library/src/components/ProgressGroup/progressGroupStyles.tsx
import styled from "styled-components";

export const ProgressGroupContainer = styled.div`
  height: auto;
  display: flex;
  justify-content: space-between;
`;

export const ProgressGroupItemContanier = styled.div<{ isActive: boolean }>`
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  position: relative;
  min-width: 230px;
  border-bottom: 4px solid transparent;

  ${({ isActive }) =>
    isActive &&
    `
      border-bottom: 4px solid var(--cora-green);
  `}

  @keyframes bounce {
    0% {
      transform: translateY(0);
      border: 2px solid var(--cora-orange);
    }
    20% {
      transform: translateY(-5px);
    }
    40% {
      transform: translateY(0);
    }
    60% {
      transform: translateY(-2px);
    }
    80% {
      transform: translateY(0);
    }
    100% {
      border: none;
    }
  }
`;

export const ProgressItem = styled.div`
  display: flex;
  justify-content: center;
  position: relative;
`;

export const ProgressGroupLine = styled.div<{ isComplete: boolean }>`
  position: absolute;
  top: 50%;
  right: 50%;
  width: 100%;
  height: 4px;
  z-index: 1;
  background-color: var(--bs-gray-300);

  ${({ isComplete }) =>
    isComplete &&
    `
    background-color: var(--cora-green);
    `}
`;

export const ProgressButtons = styled.div`
  display: flex;
  justify-content: center;
  margin: 5px 0;

  svg {
    margin: 0 10px;
    color: var(--bs-gray-500);
    animation-duration: 1.5s;
    animation-iteration-count: 1;
    transform-origin: bottom;

    &.animate-pencil {
      animation-name: bounce;
      animation-timing-function: linear;
    }

    &:hover {
      border: 2px solid var(--cora-orange);
    }
  }
`;

export const ProgressLabel = styled.div<{ isActive: boolean }>`
  text-align: center;
  color: var(--cora-green);
  margin-bottom: 10px;

  ${({ isActive }) =>
    isActive &&
    `
    font-weight: 600;
    `}
`;

export const ProgressInfo = styled.div`
  width: 100%;
  position: absolute;
  background-color: var(--cora-green-very-light);
  top: calc(100% + 4px);
  box-sizing: border-box;
  z-index: 2;
  padding: 10px;
`;

export const ProgressInfoHeader = styled.div`
  height: 30px;
  line-height: 30px;
  font-weight: var(--cora-font-weight-bold);
  display: flex;
  justify-content: space-between;
`;

export const ProgressInfoX = styled.div`
  width: 30px;
`;

export const ProgressInfoBody = styled.div`
  height: 100px;
  overflow: hidden;
`;

export const ProgressInfoFooter = styled.div`
  display: flex;
  justify-content: flex-end;
  height: auto;
`;
