// @cora/cora-component-library/src/components/ProgressSteps/progressStep.stories.tsx
import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import ProgressSteps from ".";

export default {
  title: "Page Components/ProgressSteps",
  component: ProgressSteps,
  argTypes: {},
} as ComponentMeta<typeof ProgressSteps>;

const Template: ComponentStory<typeof ProgressSteps> = (args) => (
  <ProgressSteps {...args} />
);

const data = [
  {
    label: "Enterprise Details",
    isComplete: true,
    onClick: (e: any) => {
      console.log(e);
    },
  },
  {
    label: "Enterprise Stakeholders",
    isComplete: true,
    onClick: (e: any) => {
      console.log(e);
    },
  },
  {
    label: "Program Constraints",
    isComplete: true,
    onClick: (e: any) => {
      console.log(e);
    },
  },
  {
    label: "Lorem",
    isComplete: false,
    onClick: (e: any) => {
      console.log(e);
    },
  },
  {
    label: "Ipsum",
    isComplete: true,
    onClick: (e: any) => {
      console.log(e);
    },
  },
];

export const Default = Template.bind({});
Default.args = {
  steps: data,
  selectedIndex: 0,
};
