// @cora/cora-component-library/src/components/ToastGroup/toastGroup.stories.tsx
import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import ToastGroup from ".";
import { ToastProps } from "../Toast";

export default {
  title: "Page Components/ToastGroup",
  component: ToastGroup,
  argTypes: {},
} as ComponentMeta<typeof ToastGroup>;

const Template: ComponentStory<typeof ToastGroup> = (args) => {
  return <ToastGroup {...args} />;
};

const demoToasts: ToastProps[] = [
  {
    title: "Info Toast",
    message:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer ut purus imperdiet leo dictum ullamcorper.",
    autoHide: false,
    variant: "info",
    actionText: "UNDO",
    action: () => alert("Info Clicked"),
  },
  {
    title: "Warning Toast",
    message: "Aliquam pharetra justo vel mauris dictum",
    autoHide: false,
    variant: "warning",
    actionText: "UNDO",
    action: () => alert("Warning Clicked"),
  },
  {
    title: "Danger Toast",
    message: "Orci varius natoque penatibus et",
    autoHide: false,
    variant: "danger",
    actionText: "UNDO",
    action: () => alert("Danger Clicked"),
  },
  {
    title: "Success Toast",
    message: "Orci varius natoque penatibus et",
    autoHide: false,
    variant: "success",
    actionText: "UNDO",
    action: () => alert("Success Clicked"),
  },
];

export const Default = Template.bind({});
Default.args = {
  toasts: demoToasts,
  position: "bottom-center",
};
